root/types/edit_types_historic.js

```diff
@@ -0,0 +1,319 @@
+    +tracks: $ReadOnlyArray<{
+      +artist: ArtistT,
+      +length: number | null,
+      +name: string,
+      +position: number,
+      +recording: RecordingT,
+    }>,
+    +type: ReleaseGroupTypeT | null,
+  },
+  +edit_type: EDIT_HISTORIC_ADD_RELEASE,
+}>;
+
+declare type AddReleaseAnnotationHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +changelog: string,
+    +html: string,
+    +releases: $ReadOnlyArray<ReleaseT>,
+    +text: string,
+  },
+  +edit_type: EDIT_HISTORIC_ADD_RELEASE_ANNOTATION_T,
+}>;
+
+declare type AddTrackKVHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +artist?: ArtistT,
+    +length: number,
+    +name: string,
+    +position: number,
+    +recording: RecordingT,
+    +releases: $ReadOnlyArray<ReleaseT | null>,
+  },
+  +edit_type: EDIT_HISTORIC_ADD_TRACK_KV_T,
+}>;
+
+declare type AddTrackOldHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +artist_name?: string,
+    +name: string,
+    +position: number,
+    +releases: $ReadOnlyArray<ReleaseT | null>,
+  },
+  +edit_type: EDIT_HISTORIC_ADD_TRACK_T,
+}>;
+
+declare type ChangeArtistQualityHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +artist: ArtistT,
+    +quality: CompT<QualityT>,
+  },
+  +edit_type: EDIT_HISTORIC_CHANGE_ARTIST_QUALITY_T,
+}>;
+
+declare type ChangeReleaseArtistHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +artist: CompT<ArtistT>,
+    +releases: $ReadOnlyArray<ReleaseT>,
+  },
+  +edit_type: EDIT_HISTORIC_MAC_TO_SAC_T | EDIT_HISTORIC_SAC_TO_MAC_T,
+}>;
+
+declare type ChangeReleaseGroupHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +release_group: CompT<ReleaseGroupT>,
+    +releases: $ReadOnlyArray<ReleaseT>,
+  },
+  +edit_type: EDIT_HISTORIC_CHANGE_RELEASE_GROUP_T,
+}>;
+
+declare type ChangeReleaseQualityHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +changes: $ReadOnlyArray<{
+      +quality: CompT<QualityT>,
+      +releases: $ReadOnlyArray<ReleaseT>,
+    }>,
+  },
+  +edit_type: EDIT_HISTORIC_CHANGE_RELEASE_QUALITY_T,
+}>;
+
+declare type EditRelationshipHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +relationship: CompT<$ReadOnlyArray<RelationshipT>>,
+  },
+  +edit_type: EDIT_HISTORIC_EDIT_LINK_T,
+}>;
+
+declare type EditReleaseAttributesHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +changes: $ReadOnlyArray<{
+      +releases: $ReadOnlyArray<ReleaseT | null>,
+      +status: ReleaseStatusT | null,
+      +type: ReleaseGroupTypeT | ReleaseGroupHistoricTypeT | null,
+    }>,
+    +status: ReleaseStatusT | null,
+    +type: ReleaseGroupTypeT | ReleaseGroupHistoricTypeT | null,
+  },
+  +edit_type: EDIT_HISTORIC_EDIT_RELEASE_ATTRS_T,
+}>;
+
+declare type EditReleaseEventsHistoricEditT = $ReadOnly<{
+  ...GenericEditT,
+  +display_data: {
+    +additions: $ReadOnlyArray<OldReleaseEventT>,
```

Comment:
`Cannot create 'EditNote' element because property 'iswc' is missing in 'OldReleaseEventT' [1] but exists in  object type [2] in array element of property 'display_data.additions' of property 'edit'.`

Seems to clash with ISWCs, not sure if we should just rename to something else?